### DISCUSSION TITLE
can't buy claimblocks

### DISCUSSION THREAD

**Discussion by ghost** [GriefPrevention/GriefPrevention#1023] [Q&A] [answered]
I can't purchase claimblocks. When trying the command /buyclaim and /buyclaimblocks, the game tells me buying and selling is disabled. I have vault and essentials installed, tired to edit the config but no luck.

GriefPrevention:
  SeaLevelOverrides:
    survival: -1
    survival_nether: -1
    survival_the_end: -1
    lobby: -1
    creative: -1
    creative_nether: -1
    testing: -1
  Claims:
    Mode:
      survival_the_end: Survival
      creative_nether: Disabled
      survival: Survival
      survival_nether: Survival
      creative: Disabled
      testing: Creative
      lobby: Disabled
    PreventGlobalMonsterEggs: true
    PreventTheft: true
    ProtectCreatures: true
    PreventButtonsSwitches: true
    LockWoodenDoors: false
    LockTrapDoors: false
    LockFenceGates: true
    EnderPearlsRequireAccessTrust: true
    RaidTriggersRequireBuildTrust: true
    ProtectHorses: true
    ProtectDonkeys: true
    ProtectLlamas: true
    InitialBlocks: 100
    Claim Blocks Accrued Per Hour:
      Default: 100
    Max Accrued Claim Blocks:
      Default: 2000
    Accrued Idle Threshold: 0
    AccruedIdlePercent: 0
    AbandonReturnRatio: 1.0
    AutomaticNewPlayerClaimsRadius: 4
    ExtendIntoGroundDistance: 5
    MinimumWidth: 5
    MinimumArea: 50
    MaximumDepth: 0
    InvestigationTool: STICK
    ModificationTool: GOLDEN_SHOVEL
    Expiration:
      ChestClaimDays: 7
      UnusedClaimDays: 14
      AllClaims:
        DaysInactive: 60
        ExceptWhenOwnerHasTotalClaimBlocks: 10000
        ExceptWhenOwnerHasBonusClaimBlocks: 5000
      AutomaticNatureRestoration:
        SurvivalWorlds: false
    AllowTrappedInAdminClaims: false
    MaximumNumberOfClaimsPerPlayer: 5
    CreationRequiresWorldGuardBuildPermission: false
    VillagerTradingRequiresPermission: true
    CommandsRequiringAccessTrust: /sethome
    DeliverManuals: true
    ManualDeliveryDelaySeconds: 30
    RavagersBreakBlocks: true
    FireSpreadsInClaims: false
    FireDamagesInClaims: false
    LecternReadingRequiresAccessTrust: true
  Spam:
    Enabled: true
    LoginCooldownSeconds: 60
    LoginLogoutNotificationsPerMinute: 5
    ChatSlashCommands: /me;/global;/local
    WhisperSlashCommands: /tell;/pm;/r;/whisper;/msg
    WarningMessage: Please reduce your noise level.  Spammers will be banned.
    BanOffenders: true
    BanMessage: Banned for spam.
    AllowedIpAddresses: 1.2.3.4; 5.6.7.8
    DeathMessageCooldownSeconds: 120
    Logout Message Delay In Seconds: 0
  PvP:
    RulesEnabledInWorld:
      survival: false
      survival_nether: false
      survival_the_end: false
      lobby: false
      creative: false
      creative_nether: false
      testing: false
    ProtectFreshSpawns: true
    PunishLogout: true
    CombatTimeoutSeconds: 10
    AllowCombatItemDrop: false
    BlockedSlashCommands: /home;/vanish;/spawn;/tpa
    ProtectPlayersInLandClaims:
      PlayerOwnedClaims: true
      AdministrativeClaims: true
      AdministrativeSubdivisions: true
    AllowLavaDumpingNearOtherPlayers:
      PvPWorlds: true
      NonPvPWorlds: false
    AllowFlintAndSteelNearOtherPlayers:
      PvPWorlds: true
      NonPvPWorlds: false
    ProtectPetsOutsideLandClaims: false
  Economy:
    ClaimBlocksMaxBonus: 
    ClaimBlocksPurchaseCost: 100
    ClaimBlocksSellValue: 200
  ProtectItemsDroppedOnDeath:
    PvPWorlds: false
    NonPvPWorlds: true
  BlockLandClaimExplosions: true
  BlockSurfaceCreeperExplosions: true
  BlockSurfaceOtherExplosions: true
  LimitSkyTrees: true
  LimitTreeGrowth: false
  CheckPistonMovement: true
  LimitPistonsToLandClaims: true
  FireSpreads: false
  FireDestroys: false
  AdminsGetWhispers: true
  AdminsGetSignNotifications: true
  SmartBan: false
  Mute New Players Using Banned Words: false
  MaxPlayersPerIpAddress: 3
  Siege:
    Worlds: []
    BreakableBlocks:
    - DIRT
    - GRASS_BLOCK
    - GRASS
    - FERN
    - DEAD_BUSH
    - COBBLESTONE
    - GRAVEL
    - SAND
    - GLASS
    - GLASS_PANE
    - OAK_PLANKS
    - SPRUCE_PLANKS
    - BIRCH_PLANKS
    - JUNGLE_PLANKS
    - ACACIA_PLANKS
    - DARK_OAK_PLANKS
    - WHITE_WOOL
    - ORANGE_WOOL
    - MAGENTA_WOOL
    - LIGHT_BLUE_WOOL
    - YELLOW_WOOL
    - LIME_WOOL
    - PINK_WOOL
    - GRAY_WOOL
    - LIGHT_GRAY_WOOL
    - CYAN_WOOL
    - PURPLE_WOOL
    - BLUE_WOOL
    - BROWN_WOOL
    - GREEN_WOOL
    - RED_WOOL
    - BLACK_WOOL
    - SNOW
    DoorsOpenDelayInSeconds: 300
    CooldownEndInMinutes: 60
  EndermenMoveBlocks: false
  SilverfishBreakBlocks: false
  CreaturesTrampleCrops: false
  RabbitsEatCrops: true
  HardModeZombiesBreakDoors: false
  Database:
    URL: ''
    UserName: ''
    Password: ''
  UseBanCommand: false
  BanCommandPattern: ban %name% %reason%
  Advanced:
    fixNegativeClaimblockAmounts: true
    ClaimExpirationCheckRate: 60
    OfflinePlayer_cache_days: 90
  Abridged Logs:
    Days To Keep: 7
    Included Entry Types:
      Social Activity: true
      Suspicious Activity: true
      Administrative Activity: false
      Debug: false
      Muted Chat Messages: false

**Comment by Jikoo** [comment, marked as answer]
It looks like you messed up the economy settings section, but it's hard to tell because your indentation was destroyed by not using a code block. Please re-paste your config inside a code block so we can actually see what you're working with.

````
```yaml
<your config here>
```
````

Based on what little we have to go on, my suspicion is that you accidentally changed
```yaml
GriefPrevention:
  Claims:
    Economy:
      ClaimBlocksMaxBonus: 0
      ClaimBlocksPurchaseCost: 100
      ClaimBlocksSellValue: 200
```
into 
```yaml
GriefPrevention:
  Claims:
    Economy:
      ClaimBlocksMaxBonus:
        ClaimBlocksPurchaseCost: 100
        ClaimBlocksSellValue: 200
```
Not sure why you want to let users sell the blocks back for more than they bought them for, that can only be abused to gain infinite money.

**Comment by ghost** [comment]
> It looks like you messed up the economy settings section, but it's hard to tell because your indentation was destroyed by not using a code block. Please re-paste your config inside a code block so we can actually see what you're working with.
> 
> ```
> ```yaml
> <your config here>
> ```
> ```
> 
> Based on what little we have to go on, my suspicion is that you accidentally changed
> 
> ```yaml
> GriefPrevention:
>   Claims:
>     Economy:
>       ClaimBlocksMaxBonus: 0
>       ClaimBlocksPurchaseCost: 100
>       ClaimBlocksSellValue: 200
> ```
> 
> into
> 
> ```yaml
> GriefPrevention:
>   Claims:
>     Economy:
>       ClaimBlocksMaxBonus:
>         ClaimBlocksPurchaseCost: 100
>         ClaimBlocksSellValue: 200
> ```
> 
> Not sure why you want to let users sell the blocks back for more than they bought them for, that can only be abused to gain infinite money.

I have no idea what happened, but I restarted my server and it magically worked...When I reloads the config the economy looks right, but no idea why it won't work, it just worked with a restart. Also, I didn't mean to 200 for sell price, that was a mistake. Thanks!

**Comment by RoboMWM** [comment]
there's not much of a template for the question format. The bug report format is the one that has this info.
